kclvm/sema/src/resolver/lint.rs

```diff
@@ -0,0 +1,608 @@
+                    });
+                }
+            }
+        }
+    }
+}
+
+pub static UNUSED_IMPORT: &Lint = &Lint {
+    name: stringify!("UNUSED_IMPORT"),
+    level: Level::Warning,
+    desc: "Check for unused importstmt",
+    code: "W0411",
+    note: Some("Consider removing this statement"),
+};
+
+declare_lint_pass!(UnusedImport => [UNUSED_IMPORT]);
+
+fn record_use(name: &String, ctx: &mut LintContext) {
+    let re = Regex::new(r"[|:\[\]\{\}]").unwrap();
+    // # SchemaAttr.types, A|B, [A|B], {A|B:C}
+    let types: Vec<&str> = re.split(name).collect();
+    for t in types {
+        let t = t.to_string();
+        // name: a.b.c
+        let name: Vec<&str> = t.split(".").collect();
+        let firstname = name[0];
+        if let Some(import_names) = ctx.import_names.get(&ctx.filename) {
+            if import_names.contains_key(firstname) {
+                ctx.used_import_names
+                    .get_mut(&ctx.filename)
+                    .unwrap()
+                    .insert(firstname.to_string());
+            }
+        }
+    }
+}
+
+impl LintPass for UnusedImport {
+    fn check_module(
+        &mut self,
+        diags: &mut IndexSet<Diagnostic>,
+        ctx: &mut LintContext,
+        module: &ast::Module,
+    ) {
+        ctx.used_import_names
+            .insert(ctx.filename.clone(), IndexSet::default());
+    }
+
+    fn check_module_post(
+        &mut self,
+        diags: &mut IndexSet<Diagnostic>,
+        ctx: &mut LintContext,
+        module: &ast::Module,
+    ) {
+        let used_import_names = ctx.used_import_names.get(&ctx.filename).unwrap();
+        for stmt in &module.body {
+            if let ast::Stmt::Import(import_stmt) = &stmt.node {
+                if !used_import_names.contains(&import_stmt.name) {
+                    diags.insert(Diagnostic {
+                        level: Level::Warning,
+                        messages: (&[Message {
+                            pos: Position {
+                                filename: module.filename.clone(),
+                                line: stmt.line,
+                                column: None,
+                            },
+                            style: Style::Line,
+                            message: format!("Module '{}' imported but unused.", import_stmt.name),
+                            note: Some(
+                                UnusedImport::get_lints()[0]
+                                    .note
+                                    .unwrap()
+                                    .clone()
+                                    .to_string(),
+                            ),
+                        }])
+                            .to_vec(),
+                        code: Some(DiagnosticId::Warning(WarningKind::UnusedImportWarning)),
+                    });
+                }
+            }
+        }
+    }
+
+    fn check_identifier(
```

Comment:
Try to make check_identifier & check_schema_attr functions as native functions of identifier & schema attrs. @Peefy

Reply:
> Try to make check_identifier & check_schema_attr functions as native functions of identifier & schema attrs. @Peefy

Due to the lack of symbol table information in `LintContext`, functions such as `check_identifier`, `check_schema_attr`, and `record_use` functions are implemented in a tricky way, which is part of the symbol table implementation (we can query the variable type and whether the variable has been used. ). 

So a better way is to have the `LintContext` holds a reference to the Resolver Context and handle the lint logic.